pkg/apis/flagger/v1beta1/canary.go

```diff
@@ -137,6 +137,11 @@ type CanaryService struct {
 	// +optional
 	Hosts []string `json:"hosts,omitempty"`
 
+	// IsDelegation behaves as a delegate virtual service
+	// if enabled, the pilot env `PILOT_ENABLE_VIRTUAL_SERVICE_DELEGATE` must also be set to enabled.
+	// +optional
+	IsDelegation bool `json:"isDelegation,omitempty"`
```

Comment:
```suggestion
	// If enabled, Flagger would generate Istio VirtualServices without hosts and gateway, 
	// making the service compatible with Istio delegation. Note that pilot env 
	//` PILOT_ENABLE_VIRTUAL_SERVICE_DELEGATE` must also be set.
	// +optional
	Delegation bool `json:"Delegation,omitempty"`
```

Reply:
@KeisukeYamashita can you please add a section on Istio delegation to this docs please: https://github.com/weaveworks/flagger/blob/master/docs/gitbook/faq.md#istio-routing